cub/test/catch2_test_device_partition_flagged.cu

```diff
@@ -170,7 +170,8 @@ CUB_TEST("DevicePartition::Flagged does not change input", "[device][partition_f
 
   thrust::device_vector<int> flags(num_items);
   c2h::gen(CUB_SEED(1), flags, 0, 1);
-  const int num_selected = thrust::count(flags.begin(), flags.end(), 1);
+
+  const int num_selected = static_cast<int>(thrust::count(flags.begin(), flags.end(), 1));
```

Comment:
how about using `auto`?
```suggestion
  const auto num_selected = thrust::count(flags.begin(), flags.end(), 1);
```

Reply:
This would lead to `num_selected` being `std::size_t`, which should emit a warning in comparison below.